src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -717,98 +650,69 @@ private ProxyMethod(Method method, String sig, Class<?>[] parameterTypes,
          */
-        private void generateMethod(ClassWriter cw, String className) {
-            MethodType mt = MethodType.methodType(returnType, parameterTypes);
-            String desc = mt.toMethodDescriptorString();
-            int accessFlags = ACC_PUBLIC | ACC_FINAL;
-            if (method.isVarArgs()) accessFlags |= ACC_VARARGS;
-
-            MethodVisitor mv = cw.visitMethod(accessFlags,
-                    method.getName(), desc, null,
-                    typeNames(Arrays.asList(exceptionTypes)));
-
-            int[] parameterSlot = new int[parameterTypes.length];
-            int nextSlot = 1;
-            for (int i = 0; i < parameterSlot.length; i++) {
-                parameterSlot[i] = nextSlot;
-                nextSlot += getWordsPerType(parameterTypes[i]);
-            }
-
-            mv.visitCode();
-            Label L_startBlock = new Label();
-            Label L_endBlock = new Label();
-            Label L_RuntimeHandler = new Label();
-            Label L_ThrowableHandler = new Label();
-
-            List<Class<?>> catchList = computeUniqueCatchList(exceptionTypes);
-            if (catchList.size() > 0) {
-                for (Class<?> ex : catchList) {
-                    mv.visitTryCatchBlock(L_startBlock, L_endBlock, L_RuntimeHandler,
-                            dotToSlash(ex.getName()));
-                }
-
-                mv.visitTryCatchBlock(L_startBlock, L_endBlock, L_ThrowableHandler,
-                        JL_THROWABLE);
-            }
-            mv.visitLabel(L_startBlock);
-
-            mv.visitVarInsn(ALOAD, 0);
-            mv.visitFieldInsn(GETFIELD, JLR_PROXY, handlerFieldName,
-                    LJLR_INVOCATION_HANDLER);
-            mv.visitVarInsn(ALOAD, 0);
-            mv.visitFieldInsn(GETSTATIC, dotToSlash(className), methodFieldName,
-                    LJLR_METHOD);
-
-            if (parameterTypes.length > 0) {
-                // Create an array and fill with the parameters converting primitives to wrappers
-                emitIconstInsn(mv, parameterTypes.length);
-                mv.visitTypeInsn(Opcodes.ANEWARRAY, JL_OBJECT);
-                for (int i = 0; i < parameterTypes.length; i++) {
-                    mv.visitInsn(DUP);
-                    emitIconstInsn(mv, i);
-                    codeWrapArgument(mv, parameterTypes[i], parameterSlot[i]);
-                    mv.visitInsn(Opcodes.AASTORE);
-                }
-            } else {
-                mv.visitInsn(Opcodes.ACONST_NULL);
-            }
-
-            mv.visitMethodInsn(INVOKEINTERFACE, JLR_INVOCATION_HANDLER,
-                    "invoke",
-                    "(Ljava/lang/Object;Ljava/lang/reflect/Method;" +
-                            "[Ljava/lang/Object;)Ljava/lang/Object;", true);
+        private void generateMethod(ClassBuilder clb, ClassDesc className) {
+            MethodTypeDesc desc = MethodType.methodType(returnType, parameterTypes).describeConstable().orElseThrow();
```

Comment:
Can we convert this line to:
```
MethodTypeDesc desc = MethodTypeDesc.of(toClassDesc(returnType), Arrays.stream(parameterTypes).map(ProxyGenerator::toClassDesc).toArray(Class[]::new));
```

Mainly due to that `MethodType` creation involves a lot of unrelated process such as generating lambda forms, and we can remove the `MethodType` import as a result.

Reply:
yes, this is better solution, thanks.

Reply:
More accurately, I mean the overhead generated by this: https://github.com/openjdk/jdk/blob/1e9942aa112edca33f964db127df6c9ce41e86ff/src/java.base/share/classes/java/lang/invoke/MethodType.java#L408

That method type construction is somewhat expensive and involves extra work besides simply storing the return class and the parameter classes.